inference-engine/src/cldnn_engine/cldnn_engine.cpp

```diff
@@ -127,26 +138,62 @@ InferenceEngine::ICNNNetwork::Ptr clDNNEngine::CloneAndTransformNetwork(const In
-
-        manager.set_callback(transformations_callback);
-        manager.run_passes(nGraphFunc);
-
-        ngraph::pass::Manager ti_manager;
-        // Apply all transformations to TensorIterator body
-        ti_manager.register_pass<ngraph::pass::ApplyTransformationsToTIBody>(manager);
-        // Unroll will be called after all conversions
-        // temporarily switch back to plugin unroller from NGraph unroller until TI output names are corrected
-        // ti_manager.register_pass<ngraph::pass::UnrollTensorIterator>();
-        ti_manager.run_passes(nGraphFunc);
+        CLDNNPlugin::Config config = _impl->m_config;
+        const bool enableInt8 = config.enableInt8 && (config.lptVersion == Config::LptVersion::nGraph);
+
+        {
+            // Note: instead of running all Conversion Transformations you can make up your own transformation pipeline
+            ngraph::pass::Manager manager;
+            manager.register_pass<ngraph::pass::InitNodeInfo>();
+            // WA: ConvertPriorBox must be executed before the 1st ConstantFolding pass
+            manager.register_pass<ngraph::pass::ConvertPriorBox>();
+            manager.register_pass<ngraph::pass::CommonOptimizations>();
+            manager.register_pass<ngraph::pass::ConvertOpSet3ToOpSet2>();
+            manager.register_pass<ngraph::pass::ConvertOpSet2ToOpSet1>();
+
+            if (enableInt8) {
+                // [WA part1] Convert quantized FP16 model to FP32 to avoid possible overflow and mixed precision errors
+                manager.register_pass<ngraph::pass::ConvertPrecision>(ngraph::element::f16, ngraph::element::f32);
+            }
+
+            manager.set_callback(transformations_callback);
+            manager.run_passes(nGraphFunc);
+
+            ngraph::pass::Manager ti_manager;
+            // Apply all transformations to TensorIterator body
+            ti_manager.register_pass<ngraph::pass::ApplyTransformationsToTIBody>(manager);
+            // Unroll will be called after all conversions
+            ti_manager.register_pass<ngraph::pass::UnrollTensorIterator>();
+            ti_manager.run_passes(nGraphFunc);
+        }
+
+        using namespace ngraph::pass::low_precision;
+        if (enableInt8) {
+            auto params = LayerTransformation::Params(
+                true,  // updatePrecisions
+                LayerTransformation::QuantizedTensorAlignment::UpdateLevel,  // quantizedTensorAlignmentOnActivations
+                LayerTransformation::QuantizedTensorAlignment::None,  // quantizedTensorAlignmentOnWeights
+                true);  // supportAsymmetricQuantization
+            LowPrecisionTransformer transformer(LowPrecisionTransformer::getAllTransformations(params)
+                .add<MatMulTransformation, ngraph::opset1::MatMul>(LayerTransformation::Params(params).setSupportAsymmetricQuantization(false)));
+
+            transformer.transform(nGraphFunc);
```

Comment:
I suggest to follow our regular practice for transformation types. Please use FunctionPass as a base class for LowPrecisionTransformer.

Reply:
@slyalin approved this design right now. There are specific interfaces `IParamsManager` and `ILayerTransformationsManager` to get function and other transformations details.